internal/scheduler/scheduler.go

```diff
@@ -340,6 +340,9 @@ func Run(config *Configuration) error {
 	defer producer.Close()
 
 	scheduler := NewScheduler(producer, db)
-	scheduler.Run(app.CreateContextWithShutdown())
+	runErr := scheduler.Run(app.CreateContextWithShutdown())
```

Comment:
@severinson would this be sufficient for handling errors from scheduler?  Not sure what to do in the case that we can't run the scheduler.

Reply:
Just return the error here; don't wrap it. By capturing a stack trace at the point where the error occurs, it's clear what went wrong so there's no need to add additional messages.

Further, you should use `WithMessage` instead of `Wrap` to attach a message to an error generated from within the Armada code, since `Wrap` also attaches a stack trace. We should only attach a stack trace where the error is generated or, if originating in a library outside the Armada code base, at the point at which the error is first encountered within the Armada code.

It's the responsibility of the caller of this function to handle this error (probably by shutting down).

Reply:
@severinson - is this something that should be standard throughout the code base? If so, it would be good to document that somewhere publicly.

Reply:
I think the general principle is to capture the current state of the program at the point at which an error occurs using a stack trace. Exception-based languages (e.g., Python and Java) do that for you, but in go it has to be done manually. I think the approach we take is fairly standard and that it's sufficient to leave it implicit from the code.